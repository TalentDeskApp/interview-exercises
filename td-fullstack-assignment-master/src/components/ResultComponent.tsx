import React from 'react';
import { ResultItemType, ResultType } from '@/types';

// TODO add loading & empty state when the result is empty
// TODO scrolling & virtualization for more than 200 elements
const maxElements = 200;

export const ResultComponent = ({ result }: { result: ResultType }) => {
  const showResult = (item: ResultItemType) => {
    return `pA: ${item.pA}, pB: ${item.pB}, Sum: ${item.sum}`;
  };

  return (
    <div className="mt-8 w-full max-w-md pb-10">
      <ul className="space-y-2">
        {result.length > 0 &&
          result.slice(0, maxElements).map((item) => (
            <li
              key={`${item.pA}-${item.pB}`}
              className="bg-white p-4 rounded-lg shadow-md border border-gray-200"
            >
              <p className="text-gray-700">{showResult(item)}</p>
            </li>
          ))}
      </ul>
    </div>
  );
};
