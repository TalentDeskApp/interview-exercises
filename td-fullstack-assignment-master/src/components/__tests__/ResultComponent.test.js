import React from 'react';
import { render } from '@testing-library/react';
import { ResultComponent } from '../ResultComponent';

describe('<ResultComponent />', () => {
  it('Renders ResultComponent', () => {
    const result = [
      { pA: 0, pB: 1, sum: 2 },
      { pA: 1, pB: 2, sum: 3 },
    ];
    const { asFragment } = render(<ResultComponent result={result} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
