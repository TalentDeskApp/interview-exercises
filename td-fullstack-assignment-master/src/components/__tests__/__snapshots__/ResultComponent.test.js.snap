// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultComponent /> Renders ResultComponent 1`] = `
<DocumentFragment>
  <div
    class="mt-8 w-full max-w-md pb-10"
  >
    <ul
      class="space-y-2"
    >
      <li
        class="bg-white p-4 rounded-lg shadow-md border border-gray-200"
      >
        <p
          class="text-gray-700"
        >
          pA: 0, pB: 1, Sum: 2
        </p>
      </li>
      <li
        class="bg-white p-4 rounded-lg shadow-md border border-gray-200"
      >
        <p
          class="text-gray-700"
        >
          pA: 1, pB: 2, Sum: 3
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
